### PR TITLE
refactor: fix type errors in eliminateUnreferencedIdentifiers.ts

### PR DESCRIPTION
**Pull request by TkDodo**

### CODE REVIEW

File: packages/router-vite-plugin/src/eliminateUnreferencedIdentifiers.ts [hunk 1]
```diff
@@ -106,35 +106,6 @@ export const eliminateUnreferencedIdentifiers = (
...
-              property.remove()
-            }
-          })
-
-          if (
-            beforeCount !== referencesRemovedInThisPass &&
-            pattern.get('properties').length < 1
-          ) {
-            path.remove()
-          }
```
Comment (TkDodo) [inline packages/router-vite-plugin/src/eliminateUnreferencedIdentifiers.ts:137]: this seems to be a leftover as `path.node.id.type === 'ObjectPattern'` is checked one line above, which means this should be dead code

File: packages/router-vite-plugin/src/eliminateUnreferencedIdentifiers.ts [hunk 2]
```diff
@@ -69,16 +69,16 @@ export const eliminateUnreferencedIdentifiers = (
...
           }
         } else if (t.isObjectPattern(value)) {
-          handleObjectPattern(value as NodePath<BabelTypes.ObjectPattern>)
+          handleObjectPattern(value as any)
         }
       } else if (t.isRestElement(property.node)) {
         const argument = property.get('argument')
         if (
-          t.isIdentifier(argument) &&
+          t.isIdentifier(argument as any) &&
```
Comment (TkDodo) [inline packages/router-vite-plugin/src/eliminateUnreferencedIdentifiers.ts:81]: not sure what else to do here - I think it's more important to get `main` into a state that has no type-test errors

File: packages/react-router/src/root.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const rootRouteId = '__root__' as const
```
Comment (SeanCassiere) [inline packages/react-router/src/root.ts:1]: Looks like its just the import for this in `src/fileRoute` that needs to be removed.

Comment (SeanCassiere) [reply packages/react-router/src/root.ts:1]: Fixed these

Comment (TkDodo) [reply packages/react-router/src/root.ts:1]: Thx

**Pull request merged**